Automattic/woocommerce-payments
# Use the same method when retrieving and clearing payment methods' transient

Fixes #3217

#### Changes proposed in this Pull Request

Adjust mismatch in the transient name for payment methods between retrieving and clearing it.

We were using `WC_Payments_Customer_Service::get_upe_enabled_payment_method_ids()` when for when retrieving the customer's payment methods, and `WC_Payments_Customer_Service::get_upe_enabled_at_checkout_payment_method_ids()` when clearing the cache that stores the customer's payment methods. When UPE is disabled, the latter returns an empty array while the former returns an array containing 'card'. The value returned by this method is used to concatenate the transient name. This mismatch prevents the transient from being cleared when adding new payment methods, so they're not listed in the Payment Methods list under the shopper's My Account page.

#### Testing instructions

* Under WCPay's Payments -> Settings -> Payments accepted on checkout, disable it to disable UPE
* As a shopper, go to My Account -> Payment methods
* Click on "Add payment method"
* Add a card. Opposite to step 4 in the related issue, the payment method should show up now

-------------------


**Post merge**
